apply/run.go

```diff
@@ -83,35 +76,40 @@ func (c *ClusterArgs) SetClusterArgs() error {
 	var err error = nil
 	var flag bool
 	c.cluster.Spec.Image = c.imageName
-	c.cluster.Spec.Provider = common.BAREMETAL
 
-	if c.podCidr != "" {
-		if flag, err = IsCidrString(c.podCidr); !flag {
+	if c.runArgs.PodCidr != "" {
+		if flag, err = IsCidrString(c.runArgs.PodCidr); !flag {
 			return err
 		}
-		c.cluster.Spec.Network.PodCIDR = c.podCidr
+		c.cluster.Spec.Network.PodCIDR = c.runArgs.PodCidr
 	}
-	if c.svcCidr != "" {
-		if flag, err = IsCidrString(c.svcCidr); !flag {
+	if c.runArgs.SvcCidr != "" {
+		if flag, err = IsCidrString(c.runArgs.SvcCidr); !flag {
 			return err
 		}
-		c.cluster.Spec.Network.SvcCIDR = c.svcCidr
+		c.cluster.Spec.Network.SvcCIDR = c.runArgs.SvcCidr
 	}
-	if c.passwd != "" {
-		c.cluster.Spec.SSH.Passwd = c.passwd
+	if c.runArgs.Password != "" {
+		c.cluster.Spec.SSH.Passwd = c.runArgs.Password
 	}
-	if IsNumber(c.masterArgs) && (IsNumber(c.nodeArgs) || c.nodeArgs == "") {
-		c.cluster.Spec.Masters.Count = c.masterArgs
-		c.cluster.Spec.Nodes.Count = c.nodeArgs
-		c.cluster.Spec.Provider = common.DefaultCloudProvider
-	} else if IsIPList(c.masterArgs) && (IsIPList(c.nodeArgs) || c.nodeArgs == "") {
-		c.cluster.Spec.Masters.IPList = strings.Split(c.masterArgs, ",")
-		if c.nodeArgs != "" {
-			c.cluster.Spec.Nodes.IPList = strings.Split(c.nodeArgs, ",")
+	if IsNumber(c.runArgs.Masters) && (IsNumber(c.runArgs.Nodes) || c.runArgs.Nodes == "") {
```

Comment:
if utils.InList(c.runArgs.Provider, []string{common.AliCloud, common.CONTAINER}) {
c.cluster.Spec.Provider = c.runArgs.Provider
}
i think here we don’t  need to double check the value of c.runArgs.Provider

Reply:
If the provider is a Cloud mod but the masters or Nodes type is set to ipList,  i think need to check this.

Reply:
If the provider is a Cloud mod but the masters or Nodes type is set to ipList,  i think need to check this.